test/replicate.js

```diff
@@ -737,4 +737,23 @@ test('download available blocks on non-sparse update', async function (t) {
   await b.update()
 
   t.is(b.contiguousLength, 5)
+  t.is(b.length, b.contiguousLength)
+})
+
+test('non-sparse snapshot during replication', async function (t) {
+  const a = await create()
+  const b = await create(a.key, { sparse: false })
+
+  replicate(a, b, t)
+
+  await a.append(['a', 'b', 'c', 'd', 'e'])
+  await b.update()
+
+  const s = b.snapshot()
+
+  await a.append(['f', 'g', 'h', 'i', 'j'])
+  await s.update()
+
+  t.is(s.contiguousLength, 5)
+  t.is(s.length, s.contiguousLength)
```

Comment:
@kasperisager can you add one with partial replication also? where length will be !== contigLength

Reply:
I'm not entirely sure how I'd accomplish that when the non-sparse session continuously replicates.

Reply:
@kasperisager have a peer inbetween that is sparse, where you only replicate specific blocks over (i think i added a similar test)

Reply:
Ah, makes sense!

Reply:
@kasperisager i don't think we need any contigLength guarantees around snapshots etc etc, it's an internal detail like .fork